bundles/org.openhab.automation.jrubyscripting/src/main/java/org/openhab/binding/jrubyscripting/internal/JRubyScriptEngineConfiguration.java

```diff
@@ -0,0 +1,296 @@
+            .values().stream().collect(Collectors.groupingBy(v -> v.type));
+
+    /**
+     * Update configuration
+     * 
+     * @param config Configuration parameters to apply to ScripEngine
+     * @param factory ScriptEngineFactory to configure
+     */
+    void update(Map<String, Object> config, ScriptEngineFactory factory) {
+        logger.trace("JRuby Script Engine Configuration: {}", config);
+        config.forEach(this::processConfigValue);
+        configureScriptEngine(factory);
+    }
+
+    /**
+     * Apply configuration key/value to known configuration parameters
+     * 
+     * @param key Configuration key
+     * @param value Configuration value
+     */
+    private void processConfigValue(String key, Object value) {
+        OptionalConfigurationElement configurationElement = CONFIGURATION_PARAMETERS.get(key);
+        if (configurationElement != null) {
+            configurationElement.setValue(value.toString());
+        } else {
+            logger.debug("Ignoring unexpected configuration key: {}", key);
+        }
+    }
+
+    /**
+     * Configure the ScriptEngine
+     * 
+     * @param factory Script Engine to configure
+     */
+    void configureScriptEngine(ScriptEngineFactory factory) {
+
+        configureSystemProperties(CONFIGURATION_TYPE_MAP.getOrDefault(OptionalConfigurationElement.Type.SYSTEM_PROPERTY,
+                Collections.<OptionalConfigurationElement> emptyList()));
+
+        ScriptEngine engine = factory.getScriptEngine();
+
+        configureRubyEnvironment(CONFIGURATION_TYPE_MAP.getOrDefault(OptionalConfigurationElement.Type.RUBY_ENVIRONMENT,
+                Collections.<OptionalConfigurationElement> emptyList()), engine);
+
+        configureGems(CONFIGURATION_TYPE_MAP.getOrDefault(OptionalConfigurationElement.Type.GEM,
+                Collections.<OptionalConfigurationElement> emptyList()), engine);
+    }
+
+    /**
+     * Makes Gem home directory if it does not exist
+     */
+    private void ensureGemHomeExists() {
+
+        OptionalConfigurationElement gemHomeConfigElement = CONFIGURATION_PARAMETERS.get(GEM_HOME);
+        if (gemHomeConfigElement != null) {
+            Optional<String> gemHome = gemHomeConfigElement.getValue();
+            if (gemHome.isPresent()) {
+                File gemHomeDirectory = new File(gemHome.get());
+                if (gemHomeDirectory.exists() == false) {
+                    logger.debug("gem_home directory does not exist, creating");
+                    boolean created = gemHomeDirectory.mkdirs();
+                    if (created == false) {
+                        logger.debug("Error creating gem_home direcotry");
+                    }
+                }
+            } else {
+                logger.debug("Gem install requested without gem_home specified, not ensuring gem_home path exists");
+            }
+        }
+    }
+
+    /**
+     * Install a gems in ScriptEngine
+     * 
+     * @param gemsDirectives List of gems to install
```

Comment:
The doc references `gemsDirectives` while the actual parameter is `gemDirectives`

Reply:
The param for the method is `gemDirectives`, but the config param (and docs) are `gems`.